src/context/PostsContext.jsx

```diff
@@ -1,130 +1,149 @@
 import React from "react";
-import { createContext, useEffect, useState, useContext, useMemo} from "react";
+import { createContext, useEffect, useState, useContext, useMemo } from "react";
 import dummyItems from "../context/dummyItems.js";
 
-
 const PostsContext = createContext();
 
-
 function PostsProvider({ children }) {
- const [posts, setPosts] = useState([]);
- const [isLoading, setIsLoading] = useState(false);
- const [currentPost, setCurrentPost] = useState({});
- const [error, setError] = useState(null);
- const [activeCategories, setActiveCategories] = useState([]);
+  const [posts, setPosts] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [currentPost, setCurrentPost] = useState({});
+  const [error, setError] = useState(null);
+  const [activeCategories, setActiveCategories] = useState([]);
+  const [searchQuery, setSearchQuery] = useState("");
 
- 
   async function fetchPosts() {
     try {
       setIsLoading(true);
       setError(null);
 
-      setPosts(dummyItems);  
+      setPosts(dummyItems);
     } catch (err) {
       setError(err.message || "Failed to fetch posts");
     } finally {
       setIsLoading(false);
     }
-  };
+  }
 
-useEffect(() => {
-  fetchPosts();
-}, []);
+  useEffect(() => {
+    fetchPosts();
+  }, []);
+
+  const filteredPosts = useMemo(() => {
+    return posts.filter((post) => {
+      const matchesCategory =
+        activeCategories.length === 0 ||
+        activeCategories.some((category) =>
+          typeof post.category === "string"
+            ? post.category.toLowerCase().includes(category.toLowerCase())
+            : Array.isArray(post.category)
+            ? post.category.some(
+                (c) => category.toLowerCase() === c.toLowerCase()
+              )
+            : false
+        );
```

Comment:
Since the active categories are a subset of the categories present in the posts, you don't need to compare so openly by using `.toLowerCase()` and `.includes()`:

```suggestion
      const matchesCategory = activeCategories.some((category) =>
        Array.isArray(post.category)
          ? post.category.some((c) => c === category)
          : category === post.category
      );
```

I also removed the check for `activeCategories.length === 0` because if the array is empty, `.some()` will return `false`.

It would be good to review whether `post.category` can really have both types, string and array. It should allow just one, even if there is only one category.